sw/host/opentitantool/src/main.rs

```diff
@@ -107,11 +118,32 @@ fn parse_command_line(opts: Opts, mut args: ArgsOs) -> Result<Opts> {
     Ok(opts)
 }
 
+fn execute<I>(args: I, opts: &Opts, transport: &TransportWrapper) -> Result<()>
```

Comment:
Is there a particular reason this is factored out into a function? Is it to be able to catch `?` in the right way or something?
Probably worth documenting if so.

Reply:
Originally, I had factored this out into a function because I was approaching command sequencing by collecting all positional args and making a decision about whether they need to be split with `shellwords` or could be passed as a set to `RootCommandHierarch::from_iter`.

As it is now, I think it makes the `for command in &opts.exec` loop look less gnarly.

I'm considering adding a readline-based interactive mode in the future, in which case, this function will also be useful for dispatching those commands.

Reply:
Added a comment explaining that `execute` is a convenience function.